src/libs/SidebarUtils.js

```diff
@@ -100,11 +100,12 @@ function getOrderedReportIDs(reportIDFromRoute) {
     // Get all the display names for our reports in an easy to access property so we don't have to keep
     // re-running the logic
     const filteredReportsWithReportName = _.map(filteredReports, (report) => {
-        const personalDetailMap = OptionsListUtils.getPersonalDetailsForLogins(report.participants, personalDetails);
-        return {
-            ...report,
-            reportDisplayName: ReportUtils.getReportName(report, personalDetailMap, policies),
-        };
+        // Normally, the spread operator would be used here to clone the report and prevent the need to reassign the params.
+        // However, this code needs to be very performant to handle thousands of reports, so in the interest of speed, we're just going to disable this lint rule and add
+        // the reportDisplayName property to the report object directly.
+        // eslint-disable-next-line no-param-reassign
```

Comment:
Nifty 👍 I can't think of any downside to this, worst case is `reportDisplayName` stays on the `report` object even when it's not needed, but I can't see how that would cause any problems 👍

Reply:
I strongly disagree with allowing mutations into our code, it creates hard to debug bugs and is hard to track the states of objects.

Reply:
Now there is a second property getting mutated below this one: https://github.com/Expensify/App/blob/c32fb5a9a04423bbfb5f07326431c8de186615aa/src/libs/SidebarUtils.js#L112-L113

Reply:
I'm fine with finding a way to remove the mutations as long as the code remains performant.

Reply:
I guess that makes it a bit better because we are only mutating the copy we pass down this tree, right?

https://github.com/Expensify/App/blob/fe8aeb849bd973fe92d2ee81c68e21cdb62816bf/src/pages/home/sidebar/SidebarLinksData.js#L150-L159

Still a bad practice in react to mutate props in general, you have to follow up from where the prop is coming to verify that this is not causing more side effects and it is also hard to assure that things are getting rendered when they should down the line.

> I'm fine with finding a way to remove the mutations as long as the code remains performant.

That is fair, do you know how to reproduce the performance issues?

Reply:
The best way to ensure it doesn't regress on performance is to add some timers around this code, and then time it before/after making the change.